### PR TITLE
add arn path to ocm and user role

### PR DESCRIPTION
**Pull request by ydayagi**
[SDA-6560](https://issues.redhat.com//browse/SDA-6560)
[SDA-6576](https://issues.redhat.com/browse/SDA-6576)
https://issues.redhat.com/browse/SDA-6560
https://issues.redhat.com/browse/SDA-6576
Signed-off-by: Yaron Dayagi <ydayagi@redhat.com>

### CODE REVIEW

File: pkg/aws/helpers.go [hunk 1]
```diff
@@ -402,9 +402,9 @@ func GetRolePath(roleARN string) (string, error) {
 	return path, nil
 }
 
-func GetRoleARN(accountID string, name string) string {
+func GetRoleARN(accountID string, name string, path string) string {
```
Comment (oriAdler) [inline pkg/aws/helpers.go:405]: I guess that if the user didn't provide a path, `path` equals an empty string.
If we decide to change the default value for the path to `/` it's going to affect this function as well.

**Comment by oriAdler** [comment]
@ydayagi another approach for the case the user didn't provide a **path** flag,
is to append the `--path %s` only if `path` isn't an empty string.
Does this maybe require fewer changes, WDYT?

**Comment by ydayagi** [comment]
> @ydayagi another approach for the case the user didn't provide a **path** flag, is to append the `--path %s` (manual mode) only if `path` isn't an empty string. Does this maybe require fewer changes, WDYT?

fixed all locations. I hope

File: cmd/create/userrole/cmd.go [hunk 2]
```diff
@@ -69,6 +70,12 @@ func init() {
 		"",
 		"The ARN of the policy that is used to set the permissions boundary for the user role.",
 	)
+	flags.StringVar(
+		&args.path,
+		"path",
+		"",
+		"The path for the user role.",
```
[the remark this answers is not in the document]
Comment (ydayagi) [reply cmd/create/userrole/cmd.go:77]: fixed

File: cmd/create/ocmrole/cmd.go [hunk 3]
```diff
@@ -283,17 +307,19 @@ func buildCommands(prefix string, roleName string, permissionsBoundary string,
...
+		"\t--tags %s%s"+
+		"\t--path '%s'",
+		roleName, aws.OCMRolePolicyFile, permBoundaryFlag, iamTags, adminTags, rolePath)
 	createPolicy := fmt.Sprintf("aws iam create-policy \\\n"+
 		"\t--policy-name %s \\\n"+
 		"\t--policy-document file://sts_%s_permission_policy.json \\\n"+
-		"\t--tags %s",
-		policyName, aws.OCMRolePolicyFile, iamTags)
+		"\t--tags %s"+
+		"\t--path '%s'",
```
[the remark this answers is not in the document]
Comment (ydayagi) [reply cmd/create/ocmrole/cmd.go:317]: fixed

[the remark this answers is not in the document]
Comment (ydayagi) [reply cmd/create/ocmrole/cmd.go:317]: fixed

File: cmd/create/ocmrole/cmd.go [hunk 4]
```diff
@@ -76,6 +77,13 @@ func init() {
 		"Enable admin capabilities for the role",
 	)
 
+	flags.StringVar(
+		&args.path,
+		"path",
+		"",
+		"The role path",
```
[the remark this answers is not in the document]
Comment (ydayagi) [reply cmd/create/ocmrole/cmd.go:84]: fixed

**Comment by ydayagi** [comment]
@oriAdler I made the necessary changes

File: cmd/create/accountroles/cmd.go [hunk 5]
```diff
@@ -333,19 +333,38 @@ func buildCommands(prefix string, permissionsBoundary string, accountID string,
...
-			"\t--path %s",
-			name, file, permBoundaryFlag, iamTags, rolePath)
-		createPolicy := fmt.Sprintf("aws iam create-policy \\\n"+
-			"\t--policy-name %s \\\n"+
-			"\t--policy-document file://sts_%s_permission_policy.json"+
-			"\t--tags %s"+
-			"\t--path %s",
-			policyName, file, iamTags, policyPath)
+		createRole := ""
+		if rolePath == "" {
```
Comment (oriAdler) [inline cmd/create/accountroles/cmd.go:337]: NP
I was thinking of the following implementation:
```suggestion
		createRole := fmt.Sprintf("aws iam create-role \\\n"+
			"\t--role-name %s \\\n"+
			"\t--assume-role-policy-document file://sts_%s_trust_policy.json \\\n"+
			"%s"+
			"\t--tags %s"+
			name, file, permBoundaryFlag, iamTags)
		if rolePath != "" {
			createRole = fmt.Sprintf(createRole+"\t--path %s", rolePath)
		}
```
WDYT?

Comment (ydayagi) [reply cmd/create/accountroles/cmd.go]: > if rolePath != "" {
> 			createRole = fmt.Sprintf(createRole+"\t--path %s", rolePath)
> 		}

fixed

**Pull request merged**